src/ast.js

```diff
@@ -589,6 +618,52 @@ export class LiteralString extends Expression {
   }
 }
 
+export class LiteralTemplate extends Expression {
+  constructor(cooked, expressions, raw, func) {
+    super();
+    this.cooked = cooked;
+    this.expressions = expressions || [];
+    this.length = this.expressions.length;
+    this.tagged = func !== undefined;
+    if (this.tagged) {
+      this.cooked.raw = raw;
+      this.func = func;
+    }
+  }
+
+  evaluate(scope, lookupFunctions) {
+    const results = new Array(this.length);
+    for (let i = 0; i < this.length; i++) {
+      results[i] = this.expressions[i].evaluate(scope, lookupFunctions);
+    }
+    if (this.tagged) {
+      const func = this.func.evaluate(scope, lookupFunctions);
+      if (typeof func !== 'function') {
+        throw new Error(`${this.func} is not a function`);
+      }
+      return func.call(null, this.cooked, ...results);
```

Comment:
it should preserve execution context, instead of `null`. Test:

```js
class Test {
 a(template, exps) {
  console.log(this, template, exps);
 }
}

new Test().a`5${4}3` // this should log: Test instance, ['5', '3', raw: [...]], 4
```

Reply:
Good catch. I fixed this in my latest commit